mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -501,7 +501,13 @@ trait Params extends Identifiable with Serializable {
    */
   final def getDefault[T](param: Param[T]): Option[T] = {
     shouldOwn(param)
-    defaultParamMap.get(param)
+    try {
+      defaultParamMap.get(param)
+    } catch {
+      case e: NoSuchElementException =>
```

Comment:
I figured it was better to handle the exception case this way since the exception case should happen way less frequently. I'll switch it to string interpolation tomorrow :)